src/main/java/games/strategy/engine/lobby/server/login/LobbyLoginValidator.java

```diff
@@ -48,7 +66,17 @@ public LobbyLoginValidator() {}
     final Map<String, String> rVal = new HashMap<>();
     final HashedPassword password = new DbUserController().getPassword(userName);
     if (password != null && Strings.emptyToNull(password.value) != null) {
-      rVal.put(SALT_KEY, MD5Crypt.getSalt(MD5Crypt.MAGIC, password.value));
+      rVal.put(SALT_KEY, password.isBcrypted() ? "" : MD5Crypt.getSalt(MD5Crypt.MAGIC, password.value));
+      try {
```

Comment:
Regarding the problem I observed while testing that the user's lobby password is still being stored using `MD5Crypt`...  It looks like the root cause is because this try-catch block is inside the `if (password != null ...) {` block.  Because of this, the lobby is not sending any challenge properties for a new user, and thus the client never sends back the encrypted password in response to the challenge.